activerecord/lib/active_record/core.rb

```diff
@@ -597,8 +594,20 @@ def custom_inspect_method_defined?
         self.class.instance_method(:inspect).owner != ActiveRecord::Base.instance_method(:inspect).owner
       end
 
-      def filter_attribute?(attribute_name)
-        self.class.filter_attributes.include?(attribute_name) && !read_attribute(attribute_name).nil?
+      def inspection_filter
+        @inspection_filter ||= begin
+          mask = Class.new do
+            def to_s
+              ActiveSupport::ParameterFilter::FILTERED
+            end
+            alias inspect to_s
```

Comment:
Is this alias necessary? `mask.inspect` isn't called in our test cases.

And also, I'm not fond to reference `ActiveSupport::ParameterFilter::FILTERED` repeatedly.
How about defining `pretty_print` only?

```diff
diff --git a/activerecord/lib/active_record/core.rb b/activerecord/lib/active_record/core.rb
index 34bde3fbe9..9780258302 100644
--- a/activerecord/lib/active_record/core.rb
+++ b/activerecord/lib/active_record/core.rb
@@ -596,16 +596,10 @@ def custom_inspect_method_defined?
 
       def inspection_filter
         @inspection_filter ||= begin
-          mask = Class.new do
-            def to_s
-              ActiveSupport::ParameterFilter::FILTERED
-            end
-            alias inspect to_s
-
-            def pretty_print(pp)
-              pp.text ActiveSupport::ParameterFilter::FILTERED
-            end
-          end.new.freeze
+          mask = DelegateClass(::String).new(ActiveSupport::ParameterFilter::FILTERED)
+          def mask.pretty_print(pp)
+            pp.text __getobj__
+          end
           ActiveSupport::ParameterFilter.new(self.class.filter_attributes, mask: mask)
         end
       end
```

Reply:
I added alias for completeness but that is not necessary.
I'll update as you suggested.